### ISSUE TITLE
`create` makes all properties optional

### ISSUE THREAD

**Issue by mstykow** [alexeyraspopov/dataclass#15]
I'm not sure if this is a feature or a bug: when I create a new instance of a dataclass through `.create` all class attributes have been made optional.

**Expected:** attributes that are not explicitly marked as optional should be required so that the `create` method can raise a type error if a required property is missing.

```
class User extends Data {
  name: string;
  age: number;
}

// None of these raise type errors but should:
const user1 = User.create()
const user2 = User.create({ name: 'Ann' })

class UserWithOpt extends Data {
  name: string;
  age?: number;
}

// This would be OK:
const user3 = UserWithOpt.create({ name: 'Ann' })
```

**Comment by alexeyraspopov** [comment]
> I'm not sure if this is a feature or a bug: when I create a new instance of a dataclass through .create all class attributes have been made optional.

They are not "made optional". `.create()` receives a subset of fields, assuming that omitted keys will be using default values.

>```
>class User extends Data {
>  name: string;
>  age: number;
>}
>```

What version (or configuration) of TypeScript you're using? This code suppose to raise error of properties that have no initializer.

Prior to `v2`, optional fields were not explicitly supported because information about this fields was missing in runtime. Starting from `v2` you can define `name?: string` which will provide `string | undefined` type, just like the signature suggests.

**Comment by mstykow** [comment]
What if I don't want to assign default values but want to require certain values to be provided on initialization/creation?

I looked into your source code. They way you've TS-declared the `create` method it makes all attributes optional [here](https://github.com/alexeyraspopov/dataclass/blob/master/typings/dataclass.d.ts#L6) because you're using `Partial`. However, the fix is not that simple because there is no way to distinguish class properties from class methods in TypeScript. So if we were to remove `Partial` here the `create` method would also require all data class methods which are not part of the constructor.

Basically, to address this issue we'd have to rewrite your whole implementation. So we can just document this issue as a known limitation and close the issue.

**Comment by alexeyraspopov** [comment]
Please read through the docs, most likely your use case is not covered by the library.
